test_conformance/run_conformance.py

```diff
@@ -1,302 +1,341 @@
+            failures_this_run) + ".")
+        return failures_this_run
+    return p.returncode
+
+
+def run_tests(tests_):
+    global curent_directory
+    global process_pid
+    # Run the tests
+    failures_ = 0
+    previous_test = None
+    test_number = 1
+    for t in tests_:
+        # Print the name of the test we're running and the time
+        (test_name_, test_dir_) = t
+        if test_dir_ != previous_test:
+            print("==========   " + test_dir_)
+            log_file.write("========================================================================================\n")
+            log_file.write("========================================================================================\n")
+            log_file.write("(" + get_time() + ")     Running Tests: " + test_dir_ + "\n")
+            log_file.write("========================================================================================\n")
+            log_file.write("========================================================================================\n")
+            previous_test = test_dir_
+        print("(" + get_time() + ")     BEGIN  " + test_name_.ljust(40) + ": "),
+        log_file.write(
+            "     ----------------------------------------------------------------------------------------\n")
+        log_file.write("     (" + get_time() + ")     Running Sub Test: " + test_name_ + "\n")
+        log_file.write(
+            "     ----------------------------------------------------------------------------------------\n")
+        log_file.flush()
+        sys.stdout.flush()
+
+        # Run the test
+        result = 0
+        start_time = time.time()
+        try:
+            process_pid = 0
+            result = run_test_checking_output(current_directory, test_dir_, log_file)
+        except KeyboardInterrupt:
+            # Catch an interrupt from the user
+            write_screen_log("\nFAILED: Execution interrupted.  Killing test process, but not aborting full test run.")
+            os.kill(process_pid, 9)
+            if sys.version_info[0] < 3:
+                answer = raw_input("Abort all tests? (y/n)")
+            else:
+                answer = input("Abort all tests? (y/n)")
+            if answer.find("y") != -1:
+                write_screen_log("\nUser chose to abort all tests.")
+                log_file.close()
+                sys.exit(-1)
+            else:
+                write_screen_log("\nUser chose to continue with other tests. Reporting this test as failed.")
+                result = 1
+        run_time = (time.time() - start_time)
+
+        # Move print the finish status
+        if result == 0:
+            print("(" + get_time() + ")     PASSED " + test_name_.ljust(40) + ": (" + str(int(run_time)).rjust(
+                3) + "s, test " + str(test_number).rjust(3) + os.sep + str(len(tests_)) + ")"),
```

Comment:
Here, also, there is a trailing `,` which needs to be substituted with `end=''`, addressed in PR #1492.